doc/develop/test/coverage.rst

```diff
@@ -148,13 +160,36 @@ the coverage data collected by ``gcovr`` tool in ``twister-out/coverage.json``.
 Other reports might be chosen with ``--coverage-tool`` and ``--coverage-formats``
 command line options.
 
+To generate code coverage report including Zephyr sources as well as your application
+code outside of Zephyr repository (see :ref:`Application Types <zephyr-app-types>`)
+call Twister from your project directory with ``--coverage-basedir $ZEPHYR_BASE``
+command line option, for example:
+
+.. code-block:: console
+
+   $ $ZEPHYR_BASE/scripts/twister --coverage -p native_sim --coverage_basedir $ZEPHYR_BASE -T your_project_dir
+
+.. note::
+
+   By default, Twister calls ``gcovr`` tool which filters source files assuming real paths
+   are everywhere with `all symlinks resolved <gcovr_symlinks_>`_, so when your development
+   environment has directories with symlinks then, to avoid incomplete ``gcovr`` reports,
+   either your :ref:`ZEPHYR_BASE <important-build-vars>` should contain a real path,
+   or ``lcov`` tool used instead of ``gcovr`` with additional Twister command line
+   option ``--coverage-tool lcov``.
+
 The process differs for unit tests, which are built with the host
-toolchain and require a different board::
+toolchain and require a different board:
```

Comment:
If you want to keep the diffs minimal, you can set [default language](https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html#:~:text=You%20may%20however%20use%20the%20highlight%20directive%20to%20customize%20the%20default%20language%20used%20in%20your%20document%20(see%20for%20example%20how%20this%20is%20done%20at%20the%20beginning%20of%20this%20very%20document).)